docs/2.9.0/docs/query/pqs-user-guide.rst

```diff
@@ -505,988 +1780,2081 @@ Use a ``--config`` file to define multiple options or reflect an infrastructure-
-allow access to the ledger and are used directly in queries. They
-can be used in a similar manner to tables or views and allow
-users to focus on the data they wish to query, with the impact of
-offsets removed.
+   -XX:+HeapDumpOnOutOfMemoryError
+   -XX:HeapDumpPath=/store/path
 
--  ``active(name)``: active instances of the target contracts/interfaces
-   that existed at the time of the latest offset
--  ``creates(name)``: create events that occurred between the oldest and
-   latest offset
--  ``archives(name)``: archive events that occurred between the oldest
-   and latest offset
--  ``exercises(name)``: exercise events that occurred between the oldest
-   and latest offset
-
-The functions allow the user to focus on the
-templates/interfaces/choices they wish to query, without concern for
-`PostgreSQL name
-limits <https://www.postgresql.org/docs/current/sql-syntax-lexical.html#:~:text=maximum%20identifier%20length%20is%2063%20bytes>`__.
-The ``name`` parameter can be used with or without the package
-specified:
+If you wish to have more detailed logging for diagnosis, you can adjust
+the ``--logger-level`` parameter to ``Debug`` or ``Trace``. However, be
+aware that this will generate a lot of log output and may impact
+performance. Therefore it is recommended you de-scope particularly
+verbose components (such as Netty) to ``Info`` level. This can be done
+by adding the following arguments:
 
--  Fully qualified:
-   ``<package-id>:<module>:<template|interface|choice>``
--  Partially qualified: ``<module>:<template|interface|choice>``
+.. code:: text
 
+   --logger-level=Debug
+   --logger-mappings-io.netty=Info
+   --logger-mappings-io.grpc.netty=Info
 
-Query Patterns
-==============
+Setting the Ledger API queue-length can trade-off between memory usage
+and performance. The default value is 128, and can be increased to
+deliver more stable performance, at the cost of requiring additional
+memory. Note that the buffer will consume memory equal to the size of
+transactions in a rolling window of the buffer size:
 
-Several common ways to use the table functions are described in the following sections:
+.. code:: text
 
-- Use the most recent available state of the ledger
-- Query the ledger using a point in time
-- Query the ledger from a fixed offset
-- Set the oldest offset to consider
-- Set the oldest and latest offset by time value
-- Set a minimum offset for consistency
-- Use the widest available offset range for querying
+   --source-ledger-buffersize=1024
 
-These can be combined or altered for the purpose of the query.
+PostgreSQL Configuration
+------------------------
 
-Use the most recent available state of the ledger
--------------------------------------------------
+PostgreSQL is the datastore that PQS uses to store the ledger data, but
+it also needs to consider the read-load that is placed on it by ledger
+data consumers. A qualified DBA is required to tune PostgreSQL for a
```

Comment:
How about ... "A good starting point for the PostgreSQL settings is as follows:  
<<postgres settings>>

In some cases where high performance is required, a DBA will need to tune PostgreSQL for the particular workload and infrastructure."